build.gradle

```diff
@@ -1,109 +1,147 @@
+        testCompile group: 'org.bytedeco.javacpp-presets', name: 'opencv', version: '3.0.0-1.1', classifier: osdetector.classifier.replace("osx", "macosx")
     }
-}
 
-jacocoTestReport {
-    reports {
-        xml.enabled = true
-        html.enabled = true
+    mainClassName = 'edu.wpi.grip.core.Main'
+
+    jar {
+        manifest {
+            attributes 'Implementation-Title': 'GRIP CORE', 'Implementation-Version': version, 'Main-Class': mainClassName
+        }
     }
-}
-check.dependsOn jacocoTestReport
 
-allprojects {
-    tasks.withType(Javadoc) {
-        source compileJava.source
-        options.addStringOption('Xdoclint:all,-html', '-quiet')
+    shadowJar {
+        artifacts {
+            shadow(tasks.shadowJar.archivePath) {
+                builtBy shadowJar
+            }
+        }
     }
-}
 
-// IDE setup
+    sourceSets {
+        generated {
+            java {
+                /* Note: When referencing this it becomes `srcDirs` */
+                srcDir 'src/generated/java'
+            }
+        }
+    }
 
-eclipse.classpath {
-    file.whenMerged { cp ->
-        sourceSets.generated.java.srcDirs.forEach { srcDir ->
-            def source_folder = new org.gradle.plugins.ide.eclipse.model.SourceFolder(srcDir.getAbsolutePath(), "build/classes/merged")
-            if (cp.entries.find() { it.path == source_folder.path } == null)
-                cp.entries.add(source_folder)
+    task generateCodeFromSource(type: CodeGenerator) {
+        description 'Generates the Operation Wrappers for the OpenCV methods.\n To skip run with parm `-PskipGenerate`'
+        dest sourceSets.generated.java.srcDirs
+        removeExisting true
+    }
+    generateCodeFromSource.onlyIf { !project.hasProperty('skipGenerate') }
+    compileJava.source sourceSets.generated.java, sourceSets.main.java
+    compileJava.dependsOn generateCodeFromSource
+
+    // IDE setup
+    eclipse.classpath {
+        file.whenMerged { cp ->
+            sourceSets.generated.java.srcDirs.forEach { srcDir ->
+                def source_folder = new org.gradle.plugins.ide.eclipse.model.SourceFolder(srcDir.getAbsolutePath(), "build/classes/merged")
+                if (cp.entries.find() { it.path == source_folder.path } == null)
+                    cp.entries.add(source_folder)
+            }
         }
     }
-}
 
-idea.module {
-    sourceDirs += sourceSets.generated.java.srcDirs
+    idea.module {
+        sourceDirs += sourceSets.generated.java.srcDirs
+    }
+    // End IDE setup
 }
 
-// End IDE setup
+project(":ui") {
+    apply plugin: 'java'
+    apply plugin: 'application'
+    apply from: 'http://dl.bintray.com/shemnon/javafx-gradle/8.1.1/javafx.plugin'
+
+    dependencies {
+        compile project(path: ':core', configuration: 'shadow')
+        compile group: 'org.controlsfx', name: 'controlsfx', version: '8.40.10'
+        compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.4'
```

Comment:
If it needs to be moved there it can be added there later. What use could it have in the core?